wgpu-core/src/device/resource.rs

```diff
@@ -3406,3 +3413,55 @@ impl<A: HalApi> Resource<DeviceId> for Device<A> {
         &mut self.info
     }
 }
+
+pub struct SnatchableMarker;
+/// A guard that provides read access to snatchable data.
+pub type SnatchGuard<'a> = RwLockReadGuard<'a, SnatchableMarker>;
+/// A guard that allows snatching the snatchable data.
+pub type ExclusiveSnatchGuard<'a> = RwLockWriteGuard<'a, SnatchableMarker>;
```

Comment:
Once we move this into a module - we should make these opaque wrapper types. Specifically `SnatchLock` (name can be changed) being it's own type would be useful.

That way we can mark SnatchLock::new() as unsafe. It would force the user to have to think about if calling new is actually valid, and help prevent accidental shenanigans.